### PR TITLE
[WFLY-18467] ejb-security-programmatic-auth Quickstart Common Enhance…

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
…ments CY2023Q3
Issue: https://issues.redhat.com/browse/WFLY-18467

### CODE REVIEW

File: .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: WildFly ejb-security-programmatic-auth Quickstart CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - ejb-security-programmatic-auth/**'
```
Comment (emmartins) [inline .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml:7]: - 'ejb-security-programmatic-auth/**'

File: .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      - .github/workflows/quickstart_ci.yml
+
+jobs:
+  call-quickstart_ci:
+    uses: ./.github/workflows/quickstart_ci.yml
+    with:
+      QUICKSTART_PATH: ejb-security-programmatic-auth
+      TEST_PROVISIONED_SERVER: true
+      TEST_BOOTABLE_JAR: true
```
Comment (emmartins) [inline .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml:16]: remove please

File: ejb-security-programmatic-auth/pom.xml [hunk 3]
```diff
@@ -182,7 +199,67 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (emmartins) [inline ejb-security-programmatic-auth/pom.xml:202]: should be plugin management

File: ejb-security-programmatic-auth/pom.xml [hunk 4]
```diff
@@ -182,7 +199,67 @@
...
+        <profile>
+            <id>provisioned-server</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.wildfly.plugins</groupId>
+                        <artifactId>wildfly-maven-plugin</artifactId>
+                        <configuration>
+                            <feature-packs>
+                                <feature-pack>                                    <location>org.wildfly:wildfly-galleon-pack:${version.server}</location>
```
Comment (emmartins) [inline ejb-security-programmatic-auth/pom.xml:219]: please break line

File: ejb-security-programmatic-auth/src/main/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/EJBServlet.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (emmartins) [inline ejb-security-programmatic-auth/src/main/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/EJBServlet.java:1]: Please align javadoc and html content with the one you added at https://github.com/PrarthonaPaul/quickstart/blob/main/ejb-security-context-propagation/src/main/java/org/jboss/as/quickstarts/ejb_security_context_propagation/EJBServlet.java

File: ejb-security-programmatic-auth/src/main/java/webapp/WEB-INF/beans.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (emmartins) [inline ejb-security-programmatic-auth/src/main/java/webapp/WEB-INF/beans.xml:1]: not sure we need this, you are not adding any CDI injection

File: ejb-security-programmatic-auth/src/test/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/BasicRuntimeIT.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * The very basic runtime integration testing.
+ * @author Prarthona Paul
+ * @author emartins
+ */
+public class BasicRuntimeIT {
+    private static final String DEFAULT_SERVER_HOST = "http://localhost:8080/ejb-security-programmatic-auth";
+
+    @Test
+    public void testHTTPEndpointIsAvailable() throws IOException, InterruptedException, URISyntaxException {
```
Comment (emmartins) [inline ejb-security-programmatic-auth/src/test/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/BasicRuntimeIT.java:39]: Perhaps you need to add the user auth?

File: ejb-security-programmatic-auth/pom.xml [hunk 8]
```diff
@@ -182,7 +199,67 @@
...
+                                </feature-pack>
+                            </feature-packs>
+                            <layers>
+                                <!-- layers may be used to customize the server to provision -->
+                                <layer>cloud-server</layer>
+                            </layers>
```
Comment (emmartins) [inline ejb-security-programmatic-auth/pom.xml:225]: think you need to provision the cli script to change the config

File: .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml (hunk 1 above)
Comment (PrarthonaPaul) [reply .github/workflows/quickstart_ejb_security_programmatic_auth_ci.yml]: Fixed!

File: ejb-security-programmatic-auth/src/main/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/EJBServlet.java (hunk 5 above)
Comment (PrarthonaPaul) [reply ejb-security-programmatic-auth/src/main/java/org/jboss/as/quickstarts/ejb_security_programmatic_auth/EJBServlet.java:1]: fixed!

File: ejb-security-programmatic-auth/src/main/java/webapp/WEB-INF/beans.xml (hunk 6 above)
Comment (PrarthonaPaul) [reply ejb-security-programmatic-auth/src/main/java/webapp/WEB-INF/beans.xml]: removed it

File: ejb-security-programmatic-auth/pom.xml (hunk 8 above)
Comment (PrarthonaPaul) [reply ejb-security-programmatic-auth/pom.xml:226]: I added it, but when I run `mvn clean package -Pprovisioned-server`, I get this error: 
```
org.jboss.as.cli.CommandFormatException: Failed to get the list of the operation properties: "WFLYCTL0030: No resource definition is registered for address [
    ("subsystem" => "ejb3"),
    ("application-security-domain" => "quickstart-domain")
]": Failed to get the list of the operation properties: "WFLYCTL0030: No resource definition is registered for address [
    ("subsystem" => "ejb3"),
    ("application-security-domain" => "quickstart-domain")
]"
```
I did not have trouble when I added the configurations for deploying the application.

Comment (emmartins) [reply ejb-security-programmatic-auth/pom.xml:226]: you probably need to add the ejb layer besides the cloud-server

Comment (PrarthonaPaul) [reply ejb-security-programmatic-auth/pom.xml:227]: I have added it. 
Thanks!

**Comment by PrarthonaPaul** [comment]
Hello @emmartins 
I have gone through this quickstart and it works both for the standard configuration and for the integration-testing profile.

**Comment by emmartins** [comment]
merged, thank you @PrarthonaPaul

**Pull request merged**